apps/deploy-web/src/components/new-deployment/ManifestEdit.tsx

```diff
@@ -202,8 +236,12 @@ export const ManifestEdit: React.FunctionComponent<Props> = ({ editedManifest, s
     if (mode === selectedSdlEditMode) return;
 
     if (mode === "yaml") {
-      const sdl = sdlBuilderRef.current?.getSdl() as string;
-      setEditedManifest(sdl);
+      if (editedManifest) {
```

Comment:
Same thing here, it doesn't make sense to `getSdl()` from the sdlBuilder if the mode is yaml

Reply:
Yes, right.
I am not sure this part needs.
```
    if (mode === "yaml") {
      if (editedManifest) {
        setEditedManifest(editedManifest);
      } else {
        const sdl = sdlBuilderRef.current?.getSdl() as string;
        setEditedManifest(sdl);
      }
    }
```

Reply:
How about this?
```ts
 const onModeChange = (mode: "yaml" | "builder") => {
    if (mode === selectedSdlEditMode) return;
    setSelectedSdlEditMode(mode);
  };
```

Reply:
```ts
  const onModeChange = (mode: "yaml" | "builder") => {
    if (mode === selectedSdlEditMode) return;
    if (mode === "yaml") {
      setEditedManifest(editedManifest as string);
    } else {
      const sdl = sdlBuilderRef.current?.getSdl() as string;
      setEditedManifest(sdl);
    }
    setSelectedSdlEditMode(mode);
  };
```
Right, how about this?